dialer.go

```diff
@@ -198,6 +203,29 @@ func (d *Dialer) LookupPartitions(ctx context.Context, network string, address s
 	return prt, err
 }
 
+// connectTLS returns a tls.Conn that has already completed the Handshake
+func (d *Dialer) connectTLS(ctx context.Context, conn net.Conn) (tlsConn *tls.Conn, err error) {
+	tlsConn = tls.Client(conn, d.TLS)
+	errch := make(chan error)
+
+	go func() {
+		defer close(errch)
```

Comment:
Closing the channel is unnecessary since a value is always produced to the channel, there's no risk to block waiting for a value. It has no impact on the correctness of the code tho.